packages/playback-core/src/index.ts

```diff
@@ -331,12 +354,39 @@ export const loadMedia = (
     | 'liveSyncPosition'
     | 'subtitleTracks'
     | 'subtitleTrack'
+    | 'userConfig'
   >
 ) => {
   const shouldUseNative = useNative(props, mediaEl);
   const { src } = props;
   if (mediaEl && shouldUseNative) {
     if (typeof src === 'string') {
+      fetch(src)
+        .then((resp) => resp.text())
+        .then((multivariantPlaylistStr) => {
+          const mediaPlaylistUrl = multivariantPlaylistStr.split('\n').find((_line, idx, lines) => {
+            return idx && lines[idx - 1].startsWith('#EXT-X-STREAM-INF');
+          }) as string;
+
+          return fetch(mediaPlaylistUrl).then((resp) => resp.text());
+        })
+        .then((mediaPlaylistStr) => {
+          const typeLine = mediaPlaylistStr.split('\n').find((line) => line.startsWith('#EXT-X-PLAYLIST-TYPE')) ?? '';
+
+          const playlistType = typeLine.split(':')[1]?.trim() as HlsPlaylistTypes;
```

Comment:
the code here and in the hls scenario is identical. Break into a shared function?